pkg/frontend/admin_openshiftcluster_etcdcertificaterenew_test.go

```diff
@@ -0,0 +1,508 @@
+	mockSubID := "00000000-0000-0000-0000-000000000000"
+	mockTenantID := "00000000-0000-0000-0000-000000000000"
+	mockRG := "resourceGroup"
+	ctx := context.Background()
+
+	type test struct {
+		name                 string
+		resourceID           string
+		version              *configv1.ClusterVersion
+		etcdoperator         *operatorv1.Etcd
+		etcdoperatorRevisied *operatorv1.Etcd
+		etcdCO               *configv1.ClusterOperator
+		notBefore            time.Time
+		notAfter             time.Time
+		mocks                func(*test, *mock_adminactions.MockKubeActions)
+		wantStatusCode       int
+		wantResponse         []byte
+		wantError            string
+	}
+
+	for _, tt := range []*test{
+		{
+			name:       "validate cluster version is <4.9",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
+			version: &configv1.ClusterVersion{
+				Status: configv1.ClusterVersionStatus{
+					History: []configv1.UpdateHistory{
+						{
+							State:   configv1.CompletedUpdate,
+							Version: "4.11.44",
+						},
+					},
+				},
+			},
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
+					KubeGet(gomock.Any(), "ClusterVersion.config.openshift.io", "", "version").
+					Return(encodeClusterVersion(t, tt.version), nil)
+			},
+			wantStatusCode: http.StatusForbidden,
+			wantError:      "403: Forbidden: : etcd certificate renewal is not needed for cluster running version 4.9+",
+		},
+		{
+			name:       "validate etcd operator controller status",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
+			version: &configv1.ClusterVersion{
+				Status: configv1.ClusterVersionStatus{
+					History: []configv1.UpdateHistory{
+						{
+							State:   configv1.CompletedUpdate,
+							Version: "4.8.11",
+						},
+					},
+				},
+			},
+			etcdoperator: &operatorv1.Etcd{
+				Status: operatorv1.EtcdStatus{
+					StaticPodOperatorStatus: operatorv1.StaticPodOperatorStatus{
+						OperatorStatus: operatorv1.OperatorStatus{
+							Conditions: []operatorv1.OperatorCondition{
+								{
+									Type:   "EtcdCertSignerControllerDegraded",
+									Status: operatorv1.ConditionTrue,
+								},
+							},
+						},
+					},
+				},
+			},
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
+					KubeGet(gomock.Any(), "ClusterVersion.config.openshift.io", "", "version").MaxTimes(1).
+					Return(encodeClusterVersion(t, tt.version), nil)
+				k.EXPECT().
+					KubeGet(gomock.Any(), "etcd.operator.openshift.io", "", "cluster").
+					Return(encodeEtcdOperatorController(t, tt.etcdoperator), nil)
+			},
+			wantStatusCode: http.StatusInternalServerError,
+			wantError:      "500: InternalServerError: : EtcdCertSignerControllerDegraded is in state True, quiting.",
```

Comment:
Nit: 500 is usually reserved for when a server-caused error occurs. I'd be more inclined to use 422 which means "unprocessable content"